snap/component.go

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snap
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// ComponentInfo is the content of a component.yaml file.
+type ComponentInfo struct {
+	Component   ComponentId   `yaml:"component"`
+	Type        ComponentType `yaml:"type"`
+	Version     string        `yaml:"version"`
+	Summary     string        `yaml:"summary"`
+	Description string        `yaml:"description"`
+}
+
+// ComponentId contains the component name and the owner snap name.
+type ComponentId struct {
```

Comment:
looking at things like PlugRef, this maybe should be called ComponentRef, also it should probably live in the naming package?